src/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs

```diff
@@ -13,211 +13,147 @@
+        internal AsyncStreamReader(Stream stream, Action<string> callback, Encoding encoding, int bufferSize)
         {
-            Debug.Assert(process != null && stream != null && encoding != null && callback != null, "Invalid arguments!");
+            Debug.Assert(stream != null && encoding != null && callback != null, "Invalid arguments!");
             Debug.Assert(stream.CanRead, "Stream must be readable!");
             Debug.Assert(bufferSize > 0, "Invalid buffer size!");
 
-            _process = process;
             _stream = stream;
             _userCallBack = callback;
-            _encoding = encoding;
             _decoder = encoding.GetDecoder();
 
-            if (bufferSize < MinBufferSize) bufferSize = MinBufferSize;
-            _byteBuffer = new byte[bufferSize];
-            _maxCharsPerBuffer = encoding.GetMaxCharCount(bufferSize);
-            _charBuffer = new char[_maxCharsPerBuffer];
-
-            _eofEvent = new ManualResetEvent(false);
-            _messageQueue = new Queue<string>();
-        }
-
-        public void Close()
-        {
-            Dispose(true);
-        }
-
-        void IDisposable.Dispose()
-        {
-            Dispose(true);
-            GC.SuppressFinalize(this);
-        }
-
-        private void Dispose(bool disposing)
-        {
-            if (disposing)
-            {
-                if (_stream != null)
-                    _stream.Dispose();
-            }
-            if (_stream != null)
-            {
-                _stream = null;
-                _encoding = null;
-                _decoder = null;
-                _byteBuffer = null;
-                _charBuffer = null;
-            }
-
-            if (_eofEvent != null)
+            if (bufferSize < MinBufferSize)
             {
-                _eofEvent.Dispose();
-                _eofEvent = null;
+                bufferSize = MinBufferSize;
             }
-        }
+            _byteBuffer = new byte[bufferSize];
 
-        public Encoding CurrentEncoding
-        {
-            get { return _encoding; }
-        }
+            // This is the maximum number of chars we can get from one iteration in loop inside ReadBuffer.
+            // Used so ReadBuffer can tell when to copy data into a user's char[] directly, instead of our internal char[].
+            int maxCharsPerBuffer = encoding.GetMaxCharCount(bufferSize);
+            _charBuffer = new char[maxCharsPerBuffer];
 
-        public Stream BaseStream
-        {
-            get { return _stream; }
+            _cts = new CancellationTokenSource();
+            _messageQueue = new Queue<string>();
         }
 
         // User calls BeginRead to start the asynchronous read
         internal void BeginReadLine()
         {
-            if (_cancelOperation)
-            {
-                _cancelOperation = false;
-            }
-
             if (_sb == null)
             {
                 _sb = new StringBuilder(DefaultBufferSize);
```

Comment:
The constant has a comment stating that it is intended to be the buffer size in bytes, yet a StringBuilder is being allocated with this size. Given that sizeof(char) == 2, shouldn't this be `DefaultBufferSize / sizeof(char)`?